### PR TITLE
Rename to palindrom-polymer-client

### PR DESCRIPTION
**Pull request by alshakero**
All tests are passing and I added a spec that tests backward compatibility, i.e: that uses `<puppet-client />`.

### CODE REVIEW

**Comment by warpech** [comment]
Or, to `palindrom-client`? This custom element should be able to notify any partial written in any framework.

**Comment by alshakero** [comment]
I agree with `palindrom-client` much conciser and says what it does.

**Comment by tomalec** [comment]
> Or, to palindrom-client (without polymer- in the name)? 

I would say we need `polymer-` as 

> This custom element should be able to notify any partial written in any framework.

Give that we have new JS framework every day, we should not even consider that.

I would say that `palindrom-client` could do something generic vanilla way, like `juicy-html` or `imported-template` does with model binding.

----

Also please note that neither `puppet-client` nor `puppet-polymer-client` has to do anything with "partials" (as long as you mean "partial views"  or "partial view-models", not "partial HTML documents" == `DocumentFragment`)

This element only integrates Puppet/Palindrom with single Polymer's `dom-bind` element.

**Comment by warpech** [comment]
Fair enough, that is basically what I meant.

But the fact that in future we want to have cleaner `palindrom-client` does not mean that this repo cannot have such name already. 

We don't want our users to change the URL and the name of the custom element that initializes Palindrom session every few weeks.

**Comment by alshakero** [comment]
So what is the final agreement?

**Comment by warpech** [comment]
I don't have a strong opinion, please choose what you think is best. Long term we want to have a vanilla Javascript custom element called `palindrom-client`, which does not favor any framework.

If this repo is the starting point of that, then rename this repo to `palindrom-client` which is loaded into Starcounter as `/sys/palindrom-client/palindrom-client.html` and used by the name `<palindrom-client>`.

If this repo is not the starting point of that, then rename this repo to `palindrom-polymer-client` which is loaded into Starcounter as `/sys/palindrom-client/palindrom-client.html` and used by the name `<palindrom-client>`. This is more or less what we had now with `puppet-polymer-client` and it was confusing for everyone.

For me the first option is just simpler to follow.

**Comment by tomalec** [comment]
More-or-less I agree with above.

Also, I'd like to point out, that we will have to either migrate this from Polymer 1.x (-> 2.0) -> Vanilla V1, or `puppet-client` from V0 -> V1. So migration is inevitable.

**Comment by alshakero** [comment]
@tomalec that's what I did with Vanilla components. But is it possible to do this with Polymer components? What can I do with `is: 'palindrom-client'` and `<dom-module id="palindrom-client">`

**Comment by warpech** [comment]
Maybe something like this?

```js
var common = {
 /* my Polymer element definition */
}

var puppetClient = Object.assign({}, common);
puppetClient.is = "puppet-client";
Polymer(puppetClient);

var palindromClient = Object.assign({}, common);
palindromClient.is = "palindrom-client";
Polymer(palindromClient);
```

**Comment by alshakero** [comment]
@warpech I had the same idea, but I am not sure what would happen if 
```html
<dom-module id="palindrom-client">
```
and 
```js
is: 'palindrom-client'
```
didn't match. Tests are passing just fine. Should I consider it OK?

Because the docs say they should match

> To specify DOM to use for an element's local DOM, use the `<dom-module>` element. Give the `<dom-module>` an `id` attribute that **matches** its element's `is` property and put a `<template>` inside the `<dom-module>`. Polymer will automatically clone this template's contents into the element's local DOM.

**Comment by warpech** [comment]
I would actually do both `<dom-module is="palindrom-client">` and `<dom-module is="puppet-client">` separately, just share the JS config object that is used by them.

Is there any reason why Travis CI started failing?

screen shot 2017-03-17 at 12 38 58

**Comment by alshakero** [comment]
@warpech yes, `Object.assign` needs a polyfill in IE and Safari. I polyfilled it.

File: palindrom-client.html [hunk 1]
```diff
@@ -535,13 +537,15 @@
         var n = ~~Number(str);
         return String(n) === str && n >= 0;
     }
+    /* IE and Safari, polyfill Object.assign */
+    var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };
```
Comment (warpech) [inline palindrom-client.html]: Is that needed? We don't support IE. `Object.assign` is available in Safari 9. We support only current versions of Edge, Safari, Firefox and Chrome.

**Comment by warpech** [comment]
Can we change Travis to not test in unsupported browsers?

**Comment by alshakero** [comment]
@warpech even though I used [SauceLabs Configurator](https://wiki.saucelabs.com/display/DOCS/Platform+Configurator), Edge is still timing out. Should I dig deeper or remove it?

**Pull request merged**